### PR TITLE
Planthunt

### PR DESCRIPTION
**Pull request by Zephyr75**

### CODE REVIEW

File: app/src/main/java/ch/epfl/sdp/healthplay/database/Plant.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+public class Plant {
+    public String imagePath, name, date;
+
+    public Plant(String imagePath, String name, String date){
+        this.imagePath = imagePath;
+        this.name = name;
+        this.date = date;
+    }
+}
```
Comment (Tetardfurieux) [inline app/src/main/java/ch/epfl/sdp/healthplay/database/Plant.java:11]: Attributes should be private

File: app/src/main/java/ch/epfl/sdp/healthplay/api/PlantnetApi.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    private static String readAll(Reader rd) throws IOException {
+        StringBuilder sb = new StringBuilder();
+        int cp;
+        while ((cp = rd.read()) != -1) {
+            sb.append((char) cp);
+        }
+        return sb.toString();
+    }
+
+}
```
Comment (Tetardfurieux) [inline app/src/main/java/ch/epfl/sdp/healthplay/api/PlantnetApi.java:63]: Try to use the
 /** 
*
**/ while commenting a function

File: app/src/main/java/ch/epfl/sdp/healthplay/PlanthuntCameraActivity.java [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+
+    private static Button collectionButton;
+    private static Button captureButton;
+    private File photoFile;
+
+    private static final String STORAGE_URL = "gs://health-play-9e161.appspot.com";
+    public static StorageReference storage;
+
+    FirebaseUser user;
+
```
Comment (Tetardfurieux) [inline app/src/main/java/ch/epfl/sdp/healthplay/PlanthuntCameraActivity.java:47]: Specify if user is private/public

File: app/src/main/java/ch/epfl/sdp/healthplay/PlanthuntCameraActivity.java [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+
+
+        }
+    }
+
+
+
+
+}
```
Comment (Tetardfurieux) [inline app/src/main/java/ch/epfl/sdp/healthplay/PlanthuntCameraActivity.java:174]: Well commented

**Review by Aco-Hub — COMMENTED** [review]
On create methode are too long you should split them.

File: app/src/main/java/ch/epfl/sdp/healthplay/PlanthuntCollectionActivity.java [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+    private static final String STORAGE_URL = "gs://health-play-9e161.appspot.com";
+    public static StorageReference storage;
+    private static final String uploadedUrlFirst = "https://firebasestorage.googleapis.com/v0/b/health-play-9e161.appspot.com/o/";
+    private static final String uploadedUrlSecond = "?alt=media&token=937922cf-0744-4718-8ecf-c1abdda627c8";
+
+    FirebaseUser user;
+
+    ListView listViewPlants;
+
```
Comment (Tetardfurieux) [inline app/src/main/java/ch/epfl/sdp/healthplay/PlanthuntCollectionActivity.java:43]: User and listViewPlants should be private ?

File: app/src/androidTest/java/ch/epfl/sdp/healthplay/PlanthuntCameraActivityTest.java [hunk 6]
```diff
@@ -44,15 +45,22 @@ public void pictureUrlIsCorrectlySent(){
         resultData.putExtra("data", icon);
         Instrumentation.ActivityResult result = new Instrumentation.ActivityResult(Activity.RESULT_OK, resultData);
 
-
         intending(toPackage("com.android.camera2")).respondWith(result);
```
Comment (ozan-guven) [inline app/src/androidTest/java/ch/epfl/sdp/healthplay/PlanthuntCameraActivityTest.java:48]: Please extract this "magic" String into a constant.

File: app/src/main/java/ch/epfl/sdp/healthplay/ListAdapterPlant.java [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.bumptech.glide.Glide;
+
+import java.util.ArrayList;
+
+import ch.epfl.sdp.healthplay.database.Plant;
+
+public class ListAdapterPlant extends ArrayAdapter<Plant> {
```
Comment (ozan-guven) [inline app/src/main/java/ch/epfl/sdp/healthplay/ListAdapterPlant.java:20]: Please add a class comment to this class.

File: app/src/main/java/ch/epfl/sdp/healthplay/PlanthuntCollectionActivity.java [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+                                intent.putExtra("image", collectionImages.get(position));
+                                startActivity(intent);
+                            }
+                        });
+
+                    }
+                }
+        );
+
+
```
Comment (Tetardfurieux) [inline app/src/main/java/ch/epfl/sdp/healthplay/PlanthuntCollectionActivity.java:103]: Add a comment about the purpose of this FirebaseStorage access listener

File: app/src/main/java/ch/epfl/sdp/healthplay/ListAdapterPlant.java [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+
+    public ListAdapterPlant(Context context, ArrayList<Plant> plantList){
+        super(context, R.layout.item_plant_collection, plantList);
+    }
+
+    @NonNull
+    @Override
+    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent){
```
Comment (ozan-guven) [inline app/src/main/java/ch/epfl/sdp/healthplay/ListAdapterPlant.java:28]: Some comments would be appriciated inside this method to understand what you're doing.

**Review by ozan-guven — CHANGES_REQUESTED** [review]
In general, good comment. But please review my requests.

**Review by Tetardfurieux — CHANGES_REQUESTED** [review]
Great job, couple of things to fix tho !

File: app/src/main/java/ch/epfl/sdp/healthplay/database/Plant.java [hunk 10]
```diff
@@ -0,0 +1,11 @@
+package ch.epfl.sdp.healthplay.database;
+
+public class Plant {
+    public String imagePath, name, date;
+
```
Comment (Tetardfurieux) [inline app/src/main/java/ch/epfl/sdp/healthplay/database/Plant.java:5]: Attributs should be private

File: app/src/main/java/ch/epfl/sdp/healthplay/PlanthuntCollectionActivity.java [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+        storage = FirebaseStorage.getInstance(STORAGE_URL).getReference();
+
+        List<String> collectionNames = new ArrayList<String>();
+        List<String> collectionDates = new ArrayList<String>();
+        List<String> collectionImages = new ArrayList<String>();
+        ArrayList<Plant> collectionPlants = new ArrayList<Plant>();
+
+        storage.child("Planthunt").child(user.getUid()).listAll().addOnSuccessListener(
+                new OnSuccessListener<ListResult>() {
+                    @Override
```
Comment (Tetardfurieux) [inline app/src/main/java/ch/epfl/sdp/healthplay/PlanthuntCollectionActivity.java:62]: Add a comment to explain why and how this listener is added

**Review by Tetardfurieux — COMMENTED** [review]
Fix some comments

**Review by ozan-guven — APPROVED** [review]
Thanks you did all my requests ! Keep up with the work !

**Review by Tetardfurieux — APPROVED** [review]
Thanks for the changes !

**Pull request merged**